internal/flow/flow.go

```diff
@@ -173,6 +174,21 @@ func (s *Service) Status(ctx context.Context, namespace, service string) (Status
 	}
 	defer span.Finish()
```

Comment:
This must be changed to a non-defer version as we do not return immediately after this call anymore. I can see that this is actually a bug in the previous lookups as well. Note in below trace how the `artifact spec for environment` all ends at the same time.

![image](https://user-images.githubusercontent.com/6881694/106884367-85005b80-66e1-11eb-8555-c0d117d4ebef.png)


This can be fixed for all of them in another PR though.